### PR TITLE
Allow time for all nodes to see contract before executing

### PR DESCRIPTION
**Pull request by simonmaxen-da**
### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/TransactionService.scala [hunk 1]
```diff
@@ -898,11 +898,9 @@ class TransactionService(session: LedgerSession) extends LedgerTestSuite(session
     case Participants(Participant(alpha, operator, receiver), Participant(beta, giver)) =>
       for {
         agreementFactory <- beta.create(giver, AgreementFactory(receiver, giver))
-        agreement <- alpha
-          .exerciseAndGetContract[Agreement](
-            receiver,
-            agreementFactory.exerciseAgreementFactoryAccept,
-          )
+        agreement <- eventually {
```
Comment (SamirTalwar-DA) [inline ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/TransactionService.scala]: Perhaps use `beta.waitForParties(Seq(alpha), Set(giver, receiver))` before this instead?

Comment (gerolf-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/TransactionService.scala]: I think we mostly went with wrapping in `eventually`.

Comment (stefanobaghino-da) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/TransactionService.scala]: If I'm not mistaken `waitForParties` is already part of the test framework allocating the parties for the test. The error is raised when the test is ran with the flag to not wait for parties against a ledger that doesn't expose parties to remote participants.

Comment (SamirTalwar-DA) [reply ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/TransactionService.scala]: Ooooh, now I understand. Apologies.

**Comment by simonmaxen-da** [comment]
**TODO:**
When a transaction

**Pull request merged**